src/Zicht/Itertools/twig/Extension.php

```diff
@@ -5,44 +5,59 @@
 
 namespace Zicht\Itertools\twig;
 
+use Twig\Extension\GlobalsInterface;
 use Zicht\Itertools;
 
 /**
  * Twig extension.
  */
-class Extension extends \Twig_Extension
+class Extension extends \Twig_Extension implements GlobalsInterface
 {
+    /**
+     * {@inheritDoc}
+     */
+    public function getGlobals()
+    {
+        return [
+            'itf' => new Itertools\util\Filters(),
+            'itm' => new Itertools\util\Mappings(),
+            'itr' => new Itertools\util\Reductions(),
```

Comment:
Lets discuss the most difficult thing in programming... naming things.

Example
```twig
{# before #}
{% set list = [1, 2, 3, 4] %}
{{ list|filter(filtering("in", [1, 2])) }}

{# after #}
{% set list = [1, 2, 3, 4] %}
{{ list|it.filter(itf.in([1, 2])) }}
```

Reply:
I don't see the necessity to also have `itf` `itm` and `itr` prefixes. In PHP code it now is: `Filters::in()`, so my suggestion and expectation for Twig would be:

```twig
{{ list|it.filter(it.filtering("in", [1, 2])) }}
```

Reply:
`it` returns an iterator instance, thats why you give it the data you want to iterate over.  All subsequent calls are done directly call the iterator methods: i.e. `list|it.filter.map('foo').sorted`

Your suggestion would make `filtering`, `mapping`, and `reductions` into methods of the iterable instance, but these methods would not be chainable, which all other methods are.  These methods have nothing todo with the data (i.e. the instance), they only provide helper functions for the `filter`, `map`, and `reduce` methods.

Your suggestion is definately more intuitive, perhaps we can fix this for twig only... i.e. make a global variable `it` that can:
a. be called: `it([1, 2, 3]).filter`
b. provide access to the helper methods:
  - `[1, 2, 3]|it.filter(it.filters.in)`
  - `[1, 2, 3]|it.map(it.mappings.type)`
  - `[1, 2, 3]|it.reduce(it.reductions.add)`

Let me try this out, seems like a good improvement

Reply:
Fixed!

Example
```twig
{# before #}
{% set list = [1, 2, 3, 4] %}
{{ list|filter(filtering("in", [1, 2])) }}

{# after #}
{% set list = [1, 2, 3, 4] %}
{{ list|it.filter(it.filters.in([1, 2])) }}
```

This also removed the need to use names like 'filtering', 'mapping', and 'reducing'.  Instead the names of the classes is used: 'filters', 'mapping', and 'reductions', corresponding to the classes util/Filters.php util/Mappings.php, and util/Reductions.php.  Making php and twig more consistent with eachother

Reply:
I think this can be marked as resolved

Reply:
Great solution! 👍🏻